lib/raw_net_capture.rb

```diff
@@ -41,6 +41,14 @@ def received(data)
   def sent(data)
     @raw_sent << data
   end
+
+  def headers_only!
+    separator = "\r\n\r\n"
+    if headers_end_index = @raw_received.string.index(separator)
+      @raw_received.truncate(headers_end_index+separator.length)
+    end
+    !!headers_end_index
```

Comment:
Does it not make more sense to have a method `#headers` that just returns a dup `@raw_received.truncate(headers_end_index+separator.length)` instead of actually mutating the `@raw_received`?